draft-ietf-quic-transport.md

```diff
@@ -994,6 +994,22 @@ packets sent from only one local address.  An endpoint that migrates away from a
 local address SHOULD retire all connection IDs used on that address once it no
 longer plans to use that address.
 
+An endpoint can request that its peer retire connection IDs by sending a
+NEW_CONNECTION_ID frame with an increased Retire Prior To field.  Upon receipt,
+the peer SHOULD retire the corresponding connection IDs and send the
+corresponding RETIRE_CONNECTION_ID frame in a timely manner.  Failing to do so
+can cause packets to be delayed or lost and harm connection performance as the
+original endpoint might not route those connection IDs optimally after some
+delay.
+
+The sender of the Retire Prior To field MUST keep track of the connection IDs it
+wishes to retire until it has received a corresponding RETIRE_CONNECTION_ID
+frame, with one exception: if the sender of the Retire Prior To field has used
+distinct stateless reset tokens for all of their issued connection IDs, and 3
+times the PTO has elapsed since it received an acknowledgment for its Retire
+Prior To field, then it MAY lose track of that connection ID, and respond to
+packets with that connection ID with the corresponding stateless reset token.
```

Comment:
@erickinnear can you elaborate on this? We already have the requirement that `Retire Prior To` is <= than `Sequence Number` so it cannot retire all CIDs. I'm not sure if we can require replacement of ALL previously existing CIDs.

Reply:
There's no requirement to supply as many as the peer wants, so there can't be a requirement to replace all it wants or as many as are retired either.

Reply:
We had previously discussed introducing such a requirement, but I think we might already be covered here, since we're using the existing RETIRE_CID frame for the retirement, by the "SHOULD replace when retired" that's over in that part of the text.

Reply:
(So you have 3 that get retired, for each that you retire, you get back a NEW_CID with the same RetirePriorTo as the original one)